primitives/src/vstaging/mod.rs

```diff
@@ -50,3 +50,26 @@ pub struct AsyncBackingParams {
 	/// When async backing is disabled, the only valid value is 0.
 	pub allowed_ancestry_len: u32,
 }
+
+/// Approval voting configuration parameters
+#[derive(
+	RuntimeDebug,
+	Copy,
+	Clone,
+	PartialEq,
+	Encode,
+	Decode,
+	TypeInfo,
+	serde::Serialize,
+	serde::Deserialize,
+)]
+pub struct ApprovalVotingParams {
+	/// The maximum number of candidates `approval-voting` can vote for with
+	/// a single signatures.
+	///
+	/// Setting it to 1, means we send the approval as soon as we have it available.
+	pub max_approval_coalesce_count: u32,
+	/// The maximum time we await for a candidate approval to be coalesced with
+	/// the ones for other candidate before we sign it and distribute to our peers
+	pub max_approval_coalesce_wait_millis: u32,
```

Comment:
Are you referring just to the `max_approval_coalesce_wait_millis` part or are you think that both configuration shouldn't be there ?

Reply:
The count is a hard parameter defining a boundary between validity and invalidity, which could reasonably live in the runtime or else just be an implementation constant. Though it should probably be very high if defined in the runtime.

The wait time is just tactical from a validator and more advanced strategies would be precluded by hardcoding here.

Reply:
Ok, I see, one reason I had for keeping at least having `max_approval_coalesce_count` in the runtime is that it gives us a good knob to enable v2 approvals on all validator, instead of having to basically do 2 separate releases of node to enable this logic.

But I guess just for that use-case I could go with the option were I make both `max_approval_coalesce_count` and `max_approval_coalesce_wait_ticks`  implementation constants and just have a param in the runtime `enable_approvals_v2`. Would that make more sense ?

Reply:
I haven't reviewed the changeset in-depth yet, but typically there are two approaches to enabling new features:
  1. Do a first node upgrade that adds support for the change, followed by a second after giving enough time for nodes to upgrade to the first that activates the behavior.
  2. Do a single node upgrade that adds support for the change, which is activated by a runtime switch such as a version increase.

(2) is usually better as it goes through chain governance, which it sounds like you are going for - great.

> I could go with the option were I make both max_approval_coalesce_count and max_approval_coalesce_wait_ticks implementation constants and just have a param in the runtime enable_approvals_v2. Would that make more sense 

Yep, makes sense to me, as long as the following statement is something you agree with: we don't mind coalesced bundles being arbitrarily large as long as they are valid, and don't expect that valid bundles will be of a size too large to handle in a single network message.

Reply:
>Setting a hard maximum seems valuable to protect per-block disputes load from causing overweight blocks.

Yes, we definitely need a maximum here to protect against per-block disputes.